v2/stream/new/conn.go

```diff
@@ -0,0 +1,26 @@
+package new
+
+import (
+	"context"
+	"time"
+)
+
+// conn represents a websocket connection between the server and the client
+type conn interface {
+	// close closes the websocket connection
+	close() error
```

Comment:
[minor] I would have the methods "exported". 

Reasoning: I agree that the interface itself should not be exported. However, if you wanted to use an external package that implements this interface, you couldn't. I actually can't think of a scenario when an interface method should be private.

Reply:
Who is the 'you' in this case? Who tries to use an external package that implements the interface?